lib/classes/record/adapter.php

```diff
@@ -1820,6 +1820,32 @@ public function removeChild(\record_adapter $record)
         return $this;
     }
 
+    /**
+     * @param  $storyId
+     * @return int
+     * @throws \Exception
+     */
+    public function getStoryTotalRecords($storyId)
+    {
+        if (!is_int($storyId)) {
+            throw new \Exception('Only integer for the storyId');
+        }
+
+        $sql = "SELECT count(*) as nb
+                FROM regroup s
+                INNER JOIN record r ON r.record_id = s.rid_child
+                WHERE (s.rid_parent = :storyId) AND (r.parent_record_id = 0)"
+        ;
+
```

Comment:
no. acl must be applied on children.
depending on user's rights on child collection, some children may be not visible.